cupy/polynomial/polyutils.py

```diff
@@ -0,0 +1,53 @@
+import cupy
+
+
+def trimseq(seq):
+    """Removes small polynomial series coefficients.
+
+    Args:
+        seq (cupy.ndarray): input array.
+
+    Returns:
+        cupy.ndarray: input array with trailing zeros removed. If the
+            resulting output is empty, it returns the first element
+
+    .. seealso:: :func:`numpy.polynomial.polyutils.trimseq`
+
+    """
+    if not len(seq):
+        return seq
+    ret = cupy.trim_zeros(seq, trim='b')
+    if len(ret):
+        return ret
+    return seq[:1]
+
+
+def as_series(alist, trim=True):
+    """Returns argument as a list of 1-d arrays.
+
+    Args:
+        alist (cupy.ndarray): 1-D or 2-D input array
+        trim (boolean, optional): trim trailing zeros by default
+
+    Returns:
+        list of cupy.ndarray: list of 1-D arrays.
+
+    .. seealso:: :func:`numpy.polynomial.polyutils.as_series`
+
+    """
+    arrays = []
+    for a in alist:
+        a = cupy.array(a, ndmin=1, copy=False)
+        if not a.size:
+            raise ValueError('Coefficient array is empty')
+        if a.ndim != 1:
+            raise ValueError('Coefficient array is not 1-d')
+        if trim:
+            a = trimseq(a)
+        arrays.append(a)
+    try:
+        dtype = cupy.common_type(*arrays)
```

Comment:
All the arrays in `arrays` have the same dtypes.

Reply:
Why does bool dtype fail though if not suppressed ?

Reply:
Because `TypeError: can't get common type for non-numeric array`. It's the same as NumPy's behavior.

Reply:
So, How about writing as follows?
```py
if alist.dtype == numpy.bool:
    raise ValueError(...)
dtype = cupy.common_type(alist)
```

Reply:
Won't work with `test_as_series_list`

Reply:
it fails with `test_as_series_list`

Reply:
Why?

Reply:
I think it's because of * in `dtype = cupy.common_type(*arrays)`

Reply:
How about putting it inside the loop ?

Reply:
It seems better.